pkg/block/mem/adapter.go

```diff
@@ -359,20 +455,147 @@ func (a *Adapter) ResolveNamespace(_, storageNamespace, key string, identifierTy
 }
 
 func (a *Adapter) GetRegion(_ context.Context, _, _ string) (string, error) {
-	return "", block.ErrOperationNotSupported
+	return "", fmt.Errorf("get region: %w", block.ErrOperationNotSupported)
 }
 
 func (a *Adapter) RuntimeStats() map[string]string {
 	return nil
 }
 
 func (a *Adapter) GetPresignUploadPartURL(_ context.Context, _ block.ObjectPointer, _ string, _ int) (string, error) {
-	return "", block.ErrOperationNotSupported
+	return "", fmt.Errorf("get presign upload part url: %w", block.ErrOperationNotSupported)
 }
 
-func (a *Adapter) ListParts(_ context.Context, _ block.ObjectPointer, _ string, _ block.ListPartsOpts) (*block.ListPartsResponse, error) {
-	return nil, block.ErrOperationNotSupported
+func (a *Adapter) ListParts(_ context.Context, obj block.ObjectPointer, uploadID string, opts block.ListPartsOpts) (*block.ListPartsResponse, error) {
+	a.mutex.RLock()
+	defer a.mutex.RUnlock()
+	storageID := obj.StorageID
+	if a.mpu[storageID] == nil {
+		return nil, ErrMultiPartNotFound
+	}
+	mpu, ok := a.mpu[storageID][uploadID]
+	if !ok {
+		return nil, ErrMultiPartNotFound
```

Comment:
Can we wrap this Err here and on l.474 to state _what_ went wrong, at least in the error message?

Reply:
in both cases we fail because the ID not found, no need to expose the implementation and tell anything else.
unless I didn't got what else you wanted as part of the error?!